src/context.cpp

```diff
@@ -257,24 +261,36 @@ namespace librealsense
             std::string pid_str(to_string() << std::setfill('0') << std::setw(4) << std::hex << uvc_infos.front().pid);
             std::transform(pid_str.begin(), pid_str.end(), pid_str.begin(), ::toupper);
 
+            using namespace platform;
+            auto usb_mode = raw_color_ep->get_usb_specification();
+            std::string usb_type_str("USB");
+            if (usb_spec_names.count(usb_mode) && (usb_undefined != usb_mode))
+                usb_type_str = usb_spec_names.at(usb_mode);
+
+            register_info(RS2_CAMERA_INFO_USB_TYPE_DESCRIPTOR, usb_type_str);
             register_info(RS2_CAMERA_INFO_SERIAL_NUMBER, uvc_infos.front().unique_id);
             register_info(RS2_CAMERA_INFO_PHYSICAL_PORT, uvc_infos.front().device_path);
             register_info(RS2_CAMERA_INFO_PRODUCT_ID, pid_str);
 
-            color_ep->register_processing_block(processing_block_factory::create_pbf_vector<uyvy_converter>(RS2_FORMAT_UYVY, map_supported_color_formats(RS2_FORMAT_UYVY), RS2_STREAM_COLOR));
             color_ep->register_processing_block(processing_block_factory::create_pbf_vector<yuy2_converter>(RS2_FORMAT_YUYV, map_supported_color_formats(RS2_FORMAT_YUYV), RS2_STREAM_COLOR));
-
-            raw_color_ep->try_register_pu(RS2_OPTION_BACKLIGHT_COMPENSATION);
-            raw_color_ep->try_register_pu(RS2_OPTION_BRIGHTNESS);
-            raw_color_ep->try_register_pu(RS2_OPTION_CONTRAST);
-            raw_color_ep->try_register_pu(RS2_OPTION_EXPOSURE);
-            raw_color_ep->try_register_pu(RS2_OPTION_GAMMA);
-            raw_color_ep->try_register_pu(RS2_OPTION_HUE);
-            raw_color_ep->try_register_pu(RS2_OPTION_SATURATION);
-            raw_color_ep->try_register_pu(RS2_OPTION_SHARPNESS);
-            raw_color_ep->try_register_pu(RS2_OPTION_WHITE_BALANCE);
-            raw_color_ep->try_register_pu(RS2_OPTION_ENABLE_AUTO_EXPOSURE);
-            raw_color_ep->try_register_pu(RS2_OPTION_ENABLE_AUTO_WHITE_BALANCE);
+            color_ep->register_processing_block({ {RS2_FORMAT_MJPEG} }, { {RS2_FORMAT_RGB8, RS2_STREAM_COLOR} }, []() { return std::make_shared<mjpeg_converter>(RS2_FORMAT_RGB8); });
```

Comment:
what will happened if the device doesn't support  MJPEG?

Reply:
It would be skipped in profiles enumeration ,as with the previous case of UYUV